patch-carthage-project.rb

```diff
@@ -2,23 +2,51 @@
 project_path = ARGV[0]
 project = Xcodeproj::Project.open(project_path)
 
-# Remove targets that we do not want Carthage to build, and set the deployment target to 9.0.
+# 1) Remove targets that we do not want Carthage to build, and set the deployment target to 9.0.
 carthage_targets = ["BoringSSL", "CgRPC", "SwiftGRPC", "SwiftProtobuf"]
 targets_to_remove = project.targets.select { |target| !carthage_targets.include?(target.name) }
 targets_to_remove.each do |target|
   target.remove_from_project
 end
+project.save
 
-# Add a `swift package resolve` step before building `SwiftProtobuf`.
+# 2) Add SwiftProtobuf to the build actions list
+schemePath = Xcodeproj::XCScheme.shared_data_dir(project_path) + "SwiftGRPC-Package.xcscheme"
+scheme = Xcodeproj::XCScheme.new(schemePath)
 
-swift_protobuf_target = project.targets.select { |target| target.name == "SwiftProtobuf" }[0]
-swift_protobuf_build_phases = swift_protobuf_target.build_phases
+target = project.targets.select { |target| target.name == "SwiftProtobuf" }.first
+newBuildAction = Xcodeproj::XCScheme::BuildAction::Entry.new(target)
+newBuildAction.build_for_archiving = true
+newBuildAction.build_for_profiling = true
+newBuildAction.build_for_running = true
+newBuildAction.build_for_testing = true
+scheme.build_action.add_entry(newBuildAction)
 
-swift_protobuf_target.new_shell_script_build_phase
+# 3) Adding to SwiftGRPC-Package.xcscheme a script to Pre-Actions of BuildAction.
+# Script will resolve SPM dependecies and will fix paths issues for SwiftGRPC-Carthage.xcodeproj before everytime before build action
```

Comment:
nit: Could you fix the grammar of this sentence and explain that it fixes the issues by MOVING the existing files rather than changing the Xcode project?